snf-cyclades-app/synnefo/logic/servers.py

```diff
@@ -218,6 +224,7 @@ def create_server(vm, nics, volumes, flavor, image, personality, password):
         'img_format': str(image['format']),
         'img_personality': json.dumps(personality),
         'img_properties': json.dumps(image['metadata']),
+        'auth_keys': public_key,
```

Comment:
If I understood correctly, I should check if the public key is empty and if it isn't, insert it into the creation parameters.